kafka/kafka.go

```diff
@@ -213,17 +214,37 @@ func newCPartsFromTopicPartitions(partitions []TopicPartition) (cparts *C.rd_kaf
 		defer C.free(unsafe.Pointer(ctopic))
 		rktpar := C.rd_kafka_topic_partition_list_add(cparts, ctopic, C.int32_t(part.Partition))
 		rktpar.offset = C.int64_t(part.Offset)
+
+		if part.Metadata != nil {
+			cmetadata := convertToCStringNoNull(*part.Metadata)
+			rktpar.metadata = cmetadata
+			rktpar.metadata_size = C.size_t(len(*part.Metadata))
+		}
 	}
 
 	return cparts
 }
 
+// Go string to C string without null character.
+func convertToCStringNoNull(s string) unsafe.Pointer {
```

Comment:
Just use `C.CString()` instead of adding this function, the extra nul byte doesn't matter, just make sure metadata_size does not include it.